### ISSUE TITLE
(call-interactively 'eeit) works but "M-x eeit" says "No match"

### ISSUE THREAD

**Issue by edrx** [hlissner/doom-emacs#6251]
### What did you expect to happen?

In a Doom Emacs with the package eev installed typing `M-x eeit' should run eeit.

### What actually happened?

Typing `M-x eeit' indicates that eeit isn't a command, and doesn't have a completion; typing RET after that yields "No match".

### Describe your attempts to resolve the issue

Hi, I need some hints and help to produce a "minimal broken example" that demonstrates a bug...
I am the author of a package called eev, and one of its main features is usually invoked by typing `M-x eeit', where eeit is an alias to ee-insert-test. Here are links to my presentation about that feature at the EmacsConf2021 and to the source code of eeit:

http://angg.twu.net/eev-current/eev-testblocks.el.html#eeit
http://angg.twu.net/LATEX/2021emacsconf.pdf

A friend of mine has a setting in which it is easy to run Doom Emacs, Scimax Emacs, and Vanilla Emacs. This quirk only happens in Doom Emacs.
If he tries to run M-x eeit the completion package - company, it seems - doesn't show eeit among the options, and when he types a RET after the "M-x eeit" he gets a "No match"... however, if he puts the sexps (eeit) or (call-interactively 'eeit) in a buffer and executes them with (his favorite variant of) C-x C-e, they work.
My guess is that one, or several, of the backends for company have stored the data about the available commands in a way in which "eeit" looks like something that is not a command, and also like something that doesn't have a completion.
Any hints on how to call low-level functions of company to inspect its data structures, and to inspect what it thinks that are the completions and expansions of "eeit", would probably save me hours of work... there's a big chance that I won't be able to reproduce the bug in my machine, and that I will have to send sexps to my friend, and ask him to send back their results...

Btw: I type slowly, I commit many mistakes, and my memory is quite bad. The way that I found to cope with that is by using sexps for everything - and my package, eev, is exactly about that... so hints like "thry to execute these sexps here, inspect their results, and figure out by yourself if the results are right or wrong" would be much more useful to me than hints like "type these key sequences"... =/

Thanks in advance!
Eduardo Ochs
http://angg.twu.net/#eev

### Steps to reproduce

See above.

### System Information

The bug happens in a machine with Debian, a recent Doom Emacs, and chemacs2. That machine belongs to a friend, and I am trying to 1) reproduce the bug in my machine, 2) prepare sexps with tests that my friend can execute in his machine. See the section "Describe your attempts"... for the full info.

**Comment by hlissner** [comment]
I could not reproduce any misbehavior.

1. I installed eev by adding this to `~/.doom.d/packages.el`:

```elisp
(package! eev
  :recipe (:host github
           :repo "edrx/eev"
           :files ("*.el")))
```
2. Ran `doom sync` and restarted Emacs.
3. I then loaded the file that `eeit` is defined in.
```elisp
(require 'eev-textblocks)
```
4. `M-x eeit` works as expected.

-----

There is very little information to go on here, so here are some guesses at the cause of this issue:

- You haven't loaded `eev-testblocks.el`, so Emacs doesn't know where to find the `eeit` alias.
- In your tests on Scimax and Vanilla Emacs, it is likely this package was already loaded (which `M-x package-install` would do). /Or/ your test case has a `(require 'eev-testblocks)` in it, or similar. There are other possibilities (such as the `Customize` package doing some magic to automatically load packages for you). Doom does none of this. It expects its users to load the packages they want, when they need it. Simply adding `(require 'eev-testblocks)` to your test case would "fix" this.
- The eev-testblocks.el file does not autoload `eeit`, so it won't be available before the package is loaded. You've [autoloaded eev-beginner](https://github.com/edrx/eev/blob/UTF-8/eev-beginner.el#L180), so it doesn't seem like you're a stranger to this convention. Do the same to any interactive command you want available before the package is loaded.
